web/app/controllers/Application.java

```diff
@@ -46,79 +48,121 @@
      *             routing is managed client side
      * @return {Result} build output index.html resource
      */
-    public static Result serveAsset(String path) {
+    private static Result serveAsset(String path) {
+        File indexHtml = getIndexHtml();
+        if (indexHtml != null) {
+            // Sets the Content-Disposition to inline to indicate that the browser should
+            //   not treat this as an attachment to be downloaded
+            response().setHeader("Content-Disposition", "inline");
+            return ok(indexHtml);
+        }
+
+        return internalServerError("<h1>Oops! Something's gone wrong!</h1>").as("text/html");
+    }
+
+    /**
+     * Retrieves the index.html from the build output dir
+     *
+     * @return file for index.html or null if not found
+     */
+    @Nullable
+    private static File getIndexHtml() {
         // Get the build output Html file
-        java.io.File indexHtml = new java.io.File("build/assets/index.html");
-        // Sets the Content-Disposition to inline to indicate that the browser should
-        //   not treat this as an attachment to be downloaded
-        response().setHeader("Content-Disposition", "inline");
-        return ok(indexHtml);
+        File indexHtml = new File("build/assets/index.html");
+
+        // Ensure that we have the build step completed and the file is in the right place
+        if (indexHtml.exists()) {
+            return indexHtml;
+        }
+
+        Logger.error("Could not find index at {}", indexHtml.getAbsolutePath());
+        return null;
     }
 
     public static Result healthcheck()
     {
         return ok("GOOD");
     }
 
-    @Security.Authenticated(Secured.class)
-    public static Result index()
-    {
-        Boolean isInternal = Play.application().configuration().getBoolean(LINKEDIN_INTERNAL_KEY, false);
```

Comment:
Great thanks. It's only used by the appConfig action, so it's fine as a local var. But for sure, i'll make this final. Does keeping it a local var make sense since it will only be used here? @alyiwang